README.md

```diff
@@ -1,145 +1,231 @@
+* [Limitations](#limitations)
+* [FAQ](#faq)
+  * [Can I import template from other files?](#can-i-import-template-from-other-files)
+  * [How to get syntax highlighting in IDE (or text editors)?](#how-to-get-syntax-highlighting-in-ide-or-text-editors)
+* [License](#license)
+
 ## Usage
-.babelrc
-```js
-{
-  "plugins": [
-    "transform-react-pug",
-    "transform-react-jsx"
-  ]
-}
+
+### Syntax
+
+Full information of the syntax you can find in official documentation: [pugjs.org](https://pugjs.org/).
+
+#### Basic example
+
+```jsx
+const Component = props => pug`          //- const Component = props => (
+  div                                    //-   <div>
+    if props.amount > MAX_AMOUNT         //-     {props.amount > MAX_AMOUNT ? (
+      OtherComponent(fluid crucial)      //-       <OtherComponent fluid crucial />
+    else                                 //-     ) : (
+      p You can set bigger amount ;)     //-       <p>You can set bigger amount ;)</p>
+                                         //-     )}
+    each item, index in props.items      //-     {props.items.map((item, index) => (
+      div(key=item.id)                   //-       <div key={item.id}>
+        h3 Header #{index + 1}           //-         <h3>Header {index + 1}</h3>
+        = item.body                      //-         {item.body}
+                                         //-       </div>
+                                         //-     )}
+                                         //-   </div>
+                                         //- )
+`;
+```
+
+#### How to pass functions and other primitives
+
+```jsx
+const Component = props => pug`          //- const Component = props => (
+  div                                    //-   <div>
+    button(                              //-     <button
+      type="button"                      //-       type="button"
+      onClick=props.onClick              //-       onClick={props.onClick}
+    ) Click Me                           //-     >Click Me</button>
+                                         //-
+    OtherComponent(                      //-     <OtherComponent
+      ...props.objectWithPropsForChild   //-       {...props.objectWithPropsForChild}
+      fluid                              //-       fluid
+      data-array=[1, 2, 3]               //-       data-array={[1, 2, 3]}
+    )                                    //-     />
+                                         //-   </div>
+                                         //- )
+`;
+```
+
+#### Define local variables and use javascript in attributes
+
+```jsx
+const Component = props => pug`          //- const Component = props => (
+  Fragment                               //-   <Fragment>
+    button(                              //-     <button
+      ...one                             //-       {...one}
+      ...two                             //-       {...two}
+      onClick=() => alert('Hello')       //-       onClick={() => alert('Hello')}
```

Comment:
This does not work

Reply:
Why not? I would've expected this to work. If not, does `onClick=() => { alert('Hello') }` or `onClick=(() => alert('Hello')) ` work

Reply:
@ForbesLindesay unfortunately I just left this message because it did not work in my RN project. I had no time to debug it or create an issue with detailed description.

But now I checked it on both projects: react and react-native. Works perfectly. I'll leave this line for now and will keep my eyes open on this case.